### ISSUE TITLE
Unable to update self under linux

### ISSUE THREAD

**Issue by qinzuoyan** [pingcap-incubator/tiup#46]
Because the binary `/home/mi/.tiup/bin/tiup` is locked by the running tiup process (dead lock):
```
~$ tiup version
v0.0.0 Initial Demo
~$ tiup status       
Name  Component  PID  Status  Created Time  Directory  Binary
----  ---------  ---  ------  ------------  ---------  ------
~$ tiup update --self
Error: open /home/mi/.tiup/bin/tiup: text file busy
```

**Comment by lonng** [comment]
@qinzuoyan Does patch #47 resolve this problem?

**Comment by lonng** [comment]
Closed via #47
